catalog/opensearch/catalog-opensearch-source/src/main/java/org/codice/ddf/opensearch/source/OpenSearchParserImpl.java

```diff
@@ -158,10 +159,7 @@ public void populateSpatial(
     }
 
     if (geometry != null) {
-      throw new IllegalArgumentException(
-          "Setting the "
-              + OpenSearchConstants.GEOMETRY
-              + " query parameter is currently not supported by the OpenSearchSource");
+      checkAndReplace(client, new WKTWriter().write(geometry), OpenSearchConstants.GEOMETRY, parameters);
```

Comment:
Use a thread local `WKTWriter` since it is not thread safe.
`private static final ThreadLocal<WKTWriter> WKT_WRITER_THREAD_LOCAL = ThreadLocal.withInitial(WKTWriter::new)`

Reply:
How it is being used right here is currently thread safe (Because newing instance every request) -- though not as efficient as using threadlocal.  Whether or not the possible performance overhead matters or not here is up to @lamhuy unless anyone has a strong feeling one way or another in this case.